cmd/interactions.go

```diff
@@ -39,25 +40,57 @@ type UserInteractions struct {
 	ctx *cmd.Context
 }
 
+var ReadOneChar = func(ui UserInteractions) (string, error) {
```

Comment:
Could you change NewUserInteractions to accept ReadOneChar, then you could pass a different one in tests? Then you could make NewRestoreCommand accept a `func (ctx *cmd.Context) UserInteractions` so that you could avoid the patches to test that.